pkg/cache/v3/simple.go

```diff
@@ -570,8 +585,20 @@ func (cache *snapshotCache) Fetch(ctx context.Context, request *Request) (Respon
 			return nil, &types.SkipFetchError{}
 		}
 
+		// This code is duplicated from sotw/server.go
+		state := stream.NewStreamState(len(request.ResourceNames) == 0, nil)
+		wantedResources := make(map[string]struct{}, len(request.ResourceNames))
+		for _, resourceName := range request.ResourceNames {
+			if resourceName == "*" {
+				state.SetWildcard(true)
+				continue
+			}
+			wantedResources[resourceName] = struct{}{}
+		}
+		state.SetSubscribedResourceNames(wantedResources)
```

Comment:
Couldn't this be extracted to a constructor, e.g.

```
// NewStreamState creates a new stream state based the set of requested resources.
func NewStreamState(resources []string) {
    ....
}
```

That way you don't have to duplicate the code in sotw.server.go.

Reply:
This behavior is actually different in delta, so it's a bit complex to do
The xDS protocol created a weird exception on this behavior depending on the delta vs. sotw

Reply:
I'll see how this can be updated, but not sure I can fully hide this complexity